### PR TITLE
Set zero gc period for `test_ondemand_download_large_rel` as intended

### PR DESCRIPTION
**Pull request by LizardWizzard**
It looks like it is the source of flakiness in this test.

Reuse some config parts across tests.

### CODE REVIEW

File: test_runner/regress/test_ondemand_download.py [hunk 1]
```diff
@@ -59,18 +73,14 @@ def test_ondemand_download_large_rel(
...
+        "compaction_period": None,  # reset to default
+        "compaction_threshold": "3",
+        **PAGESERVER_CONFIG,
+    }
+
     # Override defaults, to create more layers
-    tenant, _ = env.neon_cli.create_tenant(
-        conf={
-            # disable background GC
-            "gc_period": "10 m",
```
Comment (LizardWizzard) [inline test_runner/regress/test_ondemand_download.py:66]: The main change that this now become 0

**Comment by LizardWizzard** [comment]
Can we maybe simplify this config story? E g name few configs that correspond to edge cases we'd like to test and just use them as constants. I'm not sure that all parameters that are left as is should be different than the common case

**Comment by koivunej** [comment]
I think I have a draft pr wrt this test and I think I found this is not enough. On mobile rn so no link :)

**Comment by koivunej** [comment]
Also, are the submodule changes intended or by mistake?

**Comment by LizardWizzard** [comment]
> I think I have a draft pr wrt this test and I think I found this is not enough. https://github.com/neondatabase/neon/pull/3697.

Oh, I havent seen this one, I'll close this one then. 

I was also thinking about merging configs for different tests because they are quite similar.

> Also, are the submodule changes intended or by mistake?

Yep, this is not intended, was checking out back and forth and forgot to update submodules
